src/passes/opt-utils.h

```diff
@@ -56,51 +56,48 @@ inline void optimizeAfterInlining(std::unordered_set<Function*>& funcs,
 struct CallTargetReplacer : public WalkerPass<PostWalker<CallTargetReplacer>> {
   bool isFunctionParallel() override { return true; }
 
-  CallTargetReplacer(std::map<Name, Name>* replacements)
+  CallTargetReplacer(const std::map<Name, Name>& replacements)
     : replacements(replacements) {}
 
   CallTargetReplacer* create() override {
     return new CallTargetReplacer(replacements);
   }
 
   void visitCall(Call* curr) {
-    auto iter = replacements->find(curr->target);
-    if (iter != replacements->end()) {
+    auto iter = replacements.find(curr->target);
```

Comment:
Could also be maybeReplace'd

Reply:
Done using `std::function` to pass the lambda around... thoughts?

Reply:
That's snazzy. It makes `CallTargetReplacer` harder to use with a replacements dictionary (needing to re-build the logic in a lambda elsewhere), but more general (aren't tied to using a dictionary to figure out replacements at all).